index.js

```diff
@@ -27,12 +29,22 @@ function NYC (opts) {
   this._istanbul = opts.istanbul
   this.subprocessBin = opts.subprocessBin || path.resolve(__dirname, './bin/nyc.js')
   this._tempDirectory = opts.tempDirectory || './.nyc_output'
-  this.cwd = opts.cwd || process.env.NYC_CWD || process.cwd()
-  this.reporter = arrify(opts.reporter || 'text')
+
+  var cwd = opts.cwd || process.env.NYC_CWD || process.cwd()
+  var pkgPath = pkgUp.sync(cwd)
 
   // you can specify config in the nyc stanza of package.json.
-  var config = require(path.resolve(this.cwd, './package.json')).config || {}
-  config = config.nyc || {}
+  var config
+  if (pkgPath) {
+    cwd = path.dirname(pkgPath)
+    var pkg = readPkg.sync(pkgPath, {normalize: false})
+    config = pkg.nyc || (pkg.config && pkg.config.nyc)
+  }
+  config = config || {}
+
+  this.cwd = cwd
```

Comment:
Each process will do exactly the same thing, so if `cwd` does get shifted, it gets shifted the same across processes (unless a new `package.json` magically appears in the directory tree during the test - which seems unlikely).

With this, I could do:

```sh
$ cd test/foo/bar
# then the following repeatedly until the test works
$ nyc tap ./baz.js
```

instead of 

```js
$ nyc tap test/foo/bar/baz.js
```

every time.

Also, if I did the first thing (cd into a subdir), I would rather `.nyc_output` continue to be mounted in the base of my project, not whatever directory I happen to be in.

If there is no `package.json` in the hierarchy, it falls back to the old method (it puts `.nyc_output` wherever you happen to be at the time).